careers/software-engineer.md

```diff
@@ -16,6 +16,19 @@ An apprentice engineer focuses on learning the basic skills of software engineer
 
 **Collaboration:** They pair frequently. When starting a new task, they develop a plan for how to accomplish it and validate that plan with a teammate. When stuck for more than 30 minutes on a problem, they get help. They actively participate in code review with the help of a teammate.
 
+### Responsibilities
+
+* Assist other engineers in developing and maintaining software applications.
```

Comment:
This reads like the only developing they'll do is by assisting other engineers, could we modify this or add another fine for independent work of small features and bug fixes too?